Workload-Security/Integration/aws-python-amazon-inspector-integration-v2/templates/c1ws-inspector-findings-template.yaml

```diff
@@ -0,0 +1,702 @@
+                            message = f'IPS Rules assigned to {len(computers_updated)} instances for {cve_number}, for more details check "{FILE_NAME}" on "{BUCKET_NAME}" bucket'
+                        else:
+                            message = f"Already assigned rules to computers for {cve_number}"
+                    else:
+                        print("No computers found, skipping agents")
+                        message = f"No computers found for {cve_number}"
+                else:
+                    print("Not found Corresponding Filter. CVE: {}".format(cve_number))
+                    message = f"No coreesponding rule found for {cve_number}"
+
+                subject = "Auto remediation for CVEs from Inspector"
+                print("sendig sns...")
+                send_sns(topic_arn=TOPIC_ARN, subject=subject, message=message)
+
+  EventBridgeRule:
+    Type: AWS::Events::Rule
+    Properties:
+      Name: !Sub "${AWS::StackName}-inspector-finding-event"
+      EventPattern:
+        source: ["aws.inspector2"]
+        detail-type: ["Inspector2 Finding"]
+        detail:
+          status: ["ACTIVE"]
+          severity: 
+            - HIGH
+            - CRITICAL
+            - MEDIUM
+          resources:
+            type: ["AWS_EC2_INSTANCE"]
+      Targets:
+        - Arn: !GetAtt IpsRuleAutoAssignLambda.Arn
+          Id: lambda
+
+  InspectorEvent:
+    Type: AWS::Lambda::Permission
+    Properties:
+      Action: lambda:InvokeFunction
+      FunctionName: !GetAtt IpsRuleAutoAssignLambda.Arn
+      Principal: events.amazonaws.com
+      SourceArn: !GetAtt EventBridgeRule.Arn
+
+  VulnerabilityReportLambda:
+    Type: AWS::Lambda::Function
+    Properties:
+      Handler: index.lambda_handler
+      MemorySize: 128
+      Runtime: python3.9
+      Timeout: 60
+      Environment:
+        Variables:
+          cloudoneregion: !Ref CloudOneRegion
+          c1_api_key: !Ref ApiKeySecret
+          bucket_name : !Ref BucketName
+          topic_arn: !Ref SNSTopic
+          csv_filename: !Sub "${AWS::Region}-inspector_vulnerability_report.csv"
+          aws_region: !Ref AWS::Region
+      Role: !GetAtt IpsRuleAutoAssignLambdaRole.Arn
+      Code:
+        ZipFile: |
+            from __future__ import print_function
+            import os
+            import json
+            import csv
+            import sys, warnings
+            import boto3
+            from pprint import pprint
+            from tempfile import TemporaryFile
+            from urllib.request import Request, urlopen
+            from urllib.parse import urlencode
+
+
+            API_VERSION = "v1"
+            TOPIC_ARN = os.environ.get('topic_arn')
+            AWS_REGION = os.environ.get("aws_region")
+            BUCKET_NAME = os.environ.get("bucket_name")
+            FILE_NAME = os.environ.get("csv_filename")
+            C1_API_KEY = os.environ.get("c1_api_key")
```

Comment:
I suggest using a boto3 secrets manager client to retrieve the secret created in line 71 rather than an environment variable.

https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/secretsmanager.html